crates/gitbutler-core/src/virtual_branches/base.rs

```diff
@@ -355,10 +354,13 @@ pub fn update_base_branch(
         .tree()
         .context("failed to get new target commit tree")?;
 
-    let old_target_tree = repo.find_commit(target.sha)?.tree().context(format!(
-        "failed to get old target commit tree {}",
-        target.sha
-    ))?;
+    let old_target_tree = repo
+        .find_commit(target.sha.into())?
+        .tree()
+        .context(format!(
```

Comment:
When doing anything that is non-const, like `format!` which allocates, one would use `with_context(|| <expensive stuff>)` to have the cost only when needed.